tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,305 @@
+	}
+
+	conn, err := grpc.Dial(ls.endpoint, grpc.WithInsecure())
+	if err != nil {
+		return fmt.Errorf("%v (%s)", err, ls.endpoint)
+	}
+	ls.runWg = &sync.WaitGroup{}
+	ls.kvc = pb.NewKVClient(conn)
+	ls.lc = pb.NewLeaseClient(conn)
+	ls.conn = conn
+	ls.aliveLeases = make(map[int64]struct{})
+	ls.revokedLeases = make(map[int64]struct{})
+
+	return nil
+}
+
+func (ls *leaseStresser) Stress() error {
+	plog.Infof("lease Stresser %v starting ...", ls)
+	ctx, cancel := context.WithCancel(context.Background())
+	ls.cancel = cancel
+
+	go ls.run(ctx)
+	return nil
+}
+
+func (ls *leaseStresser) run(ctx context.Context) {
+	ls.runWg.Add(1)
+	defer ls.runWg.Done()
+	ls.createLeases(ctx)
+	ls.randomlyDropLeases(ctx)
+}
+
+func (ls *leaseStresser) createLeases(ctx context.Context) {
+	numOfLeasesToCreate := ls.numOfLeases - len(ls.aliveLeases)
+	var wg sync.WaitGroup
+	wg.Add(numOfLeasesToCreate)
+	for i := 0; i < numOfLeasesToCreate; i++ {
+		go func() {
+			defer wg.Done()
+			leaseID, err := ls.createLeaseAndKeepAlive(ctx)
+			if err != nil {
+				plog.Errorf("Lease creation error %v", err)
+				return
+			}
+			ls.attachKeysWithLease(ctx, leaseID)
+			ls.mu.Lock()
+			ls.aliveLeases[leaseID] = struct{}{}
+			ls.mu.Unlock()
+		}()
+	}
+	wg.Wait()
+}
+
+func (ls *leaseStresser) randomlyDropLeases(ctx context.Context) {
+	var wg sync.WaitGroup
+	wg.Add(len(ls.aliveLeases))
+	for l := range ls.aliveLeases {
+		go func(leaseID int64) {
+			defer wg.Done()
+			isLeaseDropped := ls.randomlyDropLease(ctx, leaseID)
+			if isLeaseDropped {
+				ls.mu.Lock()
+				ls.revokedLeases[leaseID] = struct{}{}
+				delete(ls.aliveLeases, leaseID)
+				ls.mu.Unlock()
+			}
+		}(l)
+	}
+	wg.Wait()
+}
+
+func (ls *leaseStresser) getLeaseByID(ctx context.Context, leaseID int64) (*pb.LeaseTimeToLiveResponse, error) {
+	ltl := &pb.LeaseTimeToLiveRequest{
+		ID:   leaseID,
+		Keys: true,
+	}
+	return ls.lc.LeaseTimeToLive(ctx, ltl, grpc.FailFast(false))
+}
+
+func (ls *leaseStresser) hasLeaseExpired(ctx context.Context, leaseID int64) (bool, error) {
+	_, err := ls.getLeaseByID(ctx, leaseID)
+
+	if grpc.ErrorDesc(err) == grpc.ErrorDesc(rpctypes.ErrGRPCLeaseNotFound) {
+		return true, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	return false, nil
+}
+
+// The keys attached to the lease has the format of "<leaseID>_<idx>" where idx is the ordering key creation
+// Since the format of keys contains about leaseID, finding keys base on "<leaseID>" prefix
+// determines whether the attached keys for a given leaseID has been deleted or not
+func (ls *leaseStresser) hasKeysAttachedToLeaseExpired(ctx context.Context, leaseID int64) (bool, error) {
+	resp, err := ls.kvc.Range(ctx, &pb.RangeRequest{
+		Key:      []byte(fmt.Sprintf("%d", leaseID)),
+		RangeEnd: []byte(clientv3.GetPrefixRangeEnd(fmt.Sprintf("%d", leaseID))),
+	}, grpc.FailFast(false))
+
+	if err != nil {
+		plog.Errorf("Retriving keys attached to lease %v error: %v \n", leaseID, err)
```

Comment:
We don't need `\n` since the log package automatically handles by inserting new line character.

Can we change to `error: %v"`?